### PR TITLE
Actions on traits and advantages

### PR DESCRIPTION
**Pull request by Gtaray**
Added actions list to martial advantages. Added effects list to both martial advantages and unit traits. 

When a martial advantage is added to a PC, the powers are copied with it. 
When a unit trait or martial advantage is added to a unit, unit effects from the source are copied to an effects list on the unit. When that unit is added to the CT, effects on that list are applied.

### CODE REVIEW

File: campaign/record_power.xml [hunk 1]
```diff
@@ -132,13 +148,19 @@
 				testdetail.setVisible(bShowTest);
 			end
 			function onDataChanged()
+				Debug.chat('onDataChanged()');
```
Comment (MeAndUnique) [inline campaign/record_power.xml:151]: Missed a Debug

File: campaign/record_martialadvantage.xml [hunk 2]
```diff
@@ -117,4 +140,103 @@
...
 	</windowclass>
+
+	<windowclass name="martialadvantages_actions">
+		<margins control="0,0,0,2" />
+		<script>
+			function onInit()    
+				update();
+			end
+
+			function onClose()
```
Comment (MeAndUnique) [inline campaign/record_martialadvantage.xml:151]: Are there plans to add something here?

File: campaign/record_martialadvantage.xml [hunk 3]
```diff
@@ -117,4 +140,103 @@
...
+			<action_button name="test_iadd">
+				<anchored to="damage_iadd" />
+				<icon normal="button_action_attack" pressed="button_action_attack_down" />
+				<target>test</target>
+			</action_button>
+
+			<list_action name="actions">
+				<datasource>.actions</datasource>
+			</list_action>
+			<!-- <scrollbar>
```
Comment (MeAndUnique) [inline campaign/record_martialadvantage.xml:211]: Probably at the point where we want to start cleaning out commented code.

File: campaign/record_power.xml [hunk 4]
```diff
@@ -109,15 +118,22 @@
 				fOnDataChanged = super.onDataChanged
 				super.onDataChanged = onDataChanged
 				local sNode = getDatabaseNode().getPath();
+
+				super.onInit();
```
Comment (MeAndUnique) [inline campaign/record_power.xml:122]: Apparently I can't leave comments on unchanged code... Do you think this has grown large enough to justify it's own script file?

File: campaign/record_unit.xml [hunk 5]
```diff
@@ -475,4 +482,91 @@
...
 	</windowclass>
+
+	<windowclass name="unit_effects">
+		<margins control="0,0,0,2" />
+		<script>
+			function onInit()    
+				update();
+			end
+
+			function onClose()
```
Comment (MeAndUnique) [inline campaign/record_unit.xml:493]: Any plans to add something here?

File: common/scripts/actionslist.lua [hunk 6]
```diff
@@ -0,0 +1,31 @@
+function addEntry(sType, bFocus)
```
Comment (MeAndUnique) [inline common/scripts/actionslist.lua:1]: Probably should have the license header for consistency. Also, spaces to tabs.

File: common/scripts/effectslist.lua [hunk 7]
```diff
@@ -0,0 +1,29 @@
+function onInit()
```
Comment (MeAndUnique) [inline common/scripts/effectslist.lua:1]: Ditto actionslist

File: scripts/manager_power_kw.lua [hunk 8]
```diff
@@ -184,6 +183,73 @@ function getDomainPowerCastAction(nodeAction)
 	return rAction;
 end
 
+-- Get action data for the actions lists on the reference sheet for martial advantages
+-- It's basicalily a copy of getPCPowerAction without the requirement for an actor, and without handling 'cast' type rolls
+function getMartialAdvantagePowerAction(nodeAction, sSubRoll)
```
Comment (MeAndUnique) [inline scripts/manager_power_kw.lua:188]: I think this is fine as is, just maybe worth keeping an eye out as a potential location for user feedback if they have other extensions that they would expect to work for martial advantage powers.

File: campaign/record_martialadvantage.xml (hunk 2 above)
Comment (Gtaray) [reply campaign/record_martialadvantage.xml]: Removed

File: campaign/record_martialadvantage.xml (hunk 3 above)
Comment (Gtaray) [reply campaign/record_martialadvantage.xml]: Yup, I cleaned out commented xml

File: campaign/record_power.xml (hunk 4 above)
Comment (Gtaray) [reply campaign/record_power.xml]: I moved most of the scripts from power classes to their own lua files.

File: campaign/record_unit.xml (hunk 5 above)
Comment (Gtaray) [reply campaign/record_unit.xml]: Relic from when I was using handlers for something.

File: common/scripts/actionslist.lua (hunk 6 above)
Comment (Gtaray) [reply common/scripts/actionslist.lua:6]: Added the license.

File: common/scripts/effectslist.lua (hunk 7 above)
Comment (Gtaray) [reply common/scripts/effectslist.lua:5]: Ditto actionlist

**Pull request merged**